.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
is this boilerplate? we have a `requirements.txt` here, do we need this if statement then?

Reply:
yeah, it is - I had the same thought. Will remove for simplicity!